llvm/lib/Target/AMDGPU/SIMachineFunctionInfo.cpp

```diff
@@ -312,6 +312,35 @@ bool SIMachineFunctionInfo::isCalleeSavedReg(const MCPhysReg *CSRegs,
   return false;
 }
 
+void SIMachineFunctionInfo::shiftSpillPhysVGPRsToLowestRange(
```

Comment:
Maybe this should be commoned with the code to compact the scratch registers

Reply:
I'm not sure that's a good choice. The only common part I see is the MRI.replaceRegWith.